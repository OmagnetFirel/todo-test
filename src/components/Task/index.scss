.taskWrapper{
    list-style: none;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    column-gap: 16px;
    padding: 0px;
    margin: 0px;
    span{
        flex: 1;
    }

    &.completed{

        span{
            text-decoration: line-through;
        }
    }
}