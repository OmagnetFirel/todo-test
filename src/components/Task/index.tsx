import React from "react";
import "./index.scss";
interface iTask {
  todo: TTodo;
  toggleComplete: (todo: TTodo) => void;
  removeTodo: (todo: TTodo) => void;
}

export type TTodo = {
  id: string;
  completed: boolean;
  title: string;
};

function Task(props: iTask) {
  const { todo, toggleComplete, removeTodo } = props;
  return (
    <li className={`taskWrapper ${todo.completed ? "completed" : ""}`}>
      <input
        type="checkbox"
        checked={todo.completed ? true : false}
        onChange={() => toggleComplete(todo)}
      />
      <span
        onClick={() => {
          toggleComplete(todo);
        }}
      >
        {todo.title}
      </span>
      <button
        onClick={() => {
          removeTodo(todo);
        }}
      >
        X
      </button>
    </li>
  );
}

export default Task;
