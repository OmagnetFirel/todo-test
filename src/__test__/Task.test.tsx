import React from "react";
import renderer from "react-test-renderer";
import Task from "../components/Task";

describe("Task", () => {
  it("should render correctly", () => {
    const component = renderer.create(<Task taskName="teste" />).toJSON();
    expect(component).toMatchSnapshot();
  });
});