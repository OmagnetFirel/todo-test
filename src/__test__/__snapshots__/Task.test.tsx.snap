// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task should render correctly 1`] = `
<li>
  teste
</li>
`;
